src/server/editorServices.ts

```diff
@@ -855,65 +858,247 @@ namespace ts.server {
+                if (configFilePresenceInfo.trackingOpenFilesMap.has(info.path)) {
+                    configFilePresenceInfo.trackingOpenFilesMap.set(info.path, false);
+                    this.logConfigFileWatchUpdate(configFileName, configFilePresenceInfo, ConfigFileWatcherStatus.InferredRootRemoved);
                 }
 
-                // even if opening config file was successful, it could still
-                // contain errors that were tolerated.
-                this.logger.info(`Opened configuration file ${configFileName}`);
-                if (errors && errors.length > 0) {
-                    return { configFileName, configFileErrors: errors };
-                }
-            }
-            else {
-                this.updateConfiguredProject(project);
+                // Close the watcher if present
+                this.closeConfigFileWatcherIfInferredRoot(configFileName, canonicalConfigFilePath,
+                    configFilePresenceInfo, /*infoIsInferredRoot*/ true, reason
+                );
             }
+        }
 
-            return { configFileName };
+        /**
+         * This is called by inferred project whenever root script info is removed from it
+         */
+        /* @internal */
+        stopWatchingConfigFilesForInferredProjectRoot(info: ScriptInfo, reason: WatcherCloseReason) {
+            this.enumerateConfigFileLocations(info, (configFileName, canonicalConfigFilePath) =>
+                this.closeWatchConfigFileForInferredProjectRoot(configFileName, canonicalConfigFilePath, info, reason)
+            );
         }
 
-        // This is different from the method the compiler uses because
-        // the compiler can assume it will always start searching in the
-        // current directory (the directory in which tsc was invoked).
-        // The server must start searching from the directory containing
-        // the newly opened file.
-        private findConfigFile(searchPath: NormalizedPath, projectRootPath?: NormalizedPath): NormalizedPath {
+        /**
+         * This function tries to search for a tsconfig.json for the given file.
+         * This is different from the method the compiler uses because
+         * the compiler can assume it will always start searching in the
+         * current directory (the directory in which tsc was invoked).
+         * The server must start searching from the directory containing
+         * the newly opened file.
+         */
+        private enumerateConfigFileLocations(info: ScriptInfo,
+            action: (configFileName: NormalizedPath, canonicalConfigFilePath: string) => boolean | void,
+            projectRootPath?: NormalizedPath) {
+            let searchPath = asNormalizedPath(getDirectoryPath(info.fileName));
+
             while (!projectRootPath || searchPath.indexOf(projectRootPath) >= 0) {
+                const canonicalSearchPath = normalizedPathToPath(searchPath, this.currentDirectory, this.toCanonicalFileName);

```

Comment:
Created https://github.com/Microsoft/TypeScript/issues/17726 to track this just to ensure that we can change this later (i am using same logic for search as before here and would like to change that in separate PR)